### ISSUE TITLE
about point covariance

### ISSUE THREAD

**Issue by deliangye** [hku-mars/VoxelMap#11]
According to the paper, covariance of point in the world:
![Screenshot from 2022-08-27 13-38-13](https://user-images.githubusercontent.com/12262760/187041691-9aa0db1d-68a6-4f9a-9836-5284d126feaa.png)

but in the codes, it seems missing  W_R_L in the second iterm, see codes:
          _cov = state.rot_end * cov * state.rot_end.transpose() +
                (-point_crossmat) * state.cov.block<3, 3>(0, 0) *
                    (-point_crossmat).transpose() +
                state.cov.block<3, 3>(3, 3);_

So, is it a bug in the codes?

thanks!

**Comment by ChongjianYUAN** [comment]
> According to the paper, covariance of point in the world: ![Screenshot from 2022-08-27 13-38-13](https://user-images.githubusercontent.com/12262760/187041691-9aa0db1d-68a6-4f9a-9836-5284d126feaa.png)
> 
> but in the codes, it seems missing W_R_L in the second iterm, see codes: _cov = state.rot_end * cov * state.rot_end.transpose() + (-point_crossmat) * state.cov.block<3, 3>(0, 0) * (-point_crossmat).transpose() + state.cov.block<3, 3>(3, 3);_
> 
> So, is it a bug in the codes?
> 
> thanks!

Yes, it is a bug. I will fix it in the next commit. Thanks for your correction!
